### ISSUE TITLE
Review code fixer for CA1861 (Avoid constant arrays)

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#6697]
The codefix here is quite concerning. Though if there is a problem, it's out of scope for the PR anyway.

If the input code was:

```csharp
public static List<object> Cases { get; } = new() { new object[0] };
```

Is there a diagnostic? How the codefix behaves? If the codefix extracted the array to a field *below* the `Cases` static property, then it's semantically different.

_Originally posted by @Youssef1313 in https://github.com/dotnet/roslyn-analyzers/pull/6696#discussion_r1233897769_

**Comment by buyaa-n** [comment]
Thank you @mavasani for [fixing](https://github.com/dotnet/roslyn-analyzers/pull/6696) [NullReferenceException from CA1861](https://github.com/dotnet/roslyn-analyzers/issues/6686)

> If the input code was:
> 
> ```csharp
> public static List<object> Cases { get; } = new() { new object[0] };
> ```
> 
> Is there a diagnostic? How the codefix behaves? If the codefix extracted the array to a field below the Cases static property, then it's semantically different.

Good point, keep warning in this case looks wrong, especially when the array length is not a constant (literal). From the [hits found in runtime repo](https://github.com/dotnet/runtime/pull/87768) I do not see a valid diagnostic (did not check all), looks most cases it will be false positives, so I am going to put up a PR that exclude these cases.

Let me know if you think there is a valid case, I could not think of any. CC @Youssef1313 @stephentoub

**Comment by stephentoub** [comment]
> If the input code was:
> public static List<object> Cases { get; } = new() { new object[0] };

That should trigger the Array.Empty analyzer.

**Comment by buyaa-n** [comment]
As mentioned in https://github.com/dotnet/roslyn-analyzers/pull/6714#discussion_r1243164789 `Array.Empty analyzer` triggered in VS
